smart-contracts/witnet-randomness-oracle/randomness-contract.md

```diff
@@ -0,0 +1,63 @@
+# WitnetRandomness utility contract
+
+The [`WitnetRandomness` contract](./api-reference.md#WitnetRandomness), also known as the "**Witnet Randomness Oracle**", is the easiest and most convenient way to generate EVM-agnostic and realiable randomness for your smart contracts. 
+
+The Witnet Foundation sustains a pre-deployed instance of the `WitnetRandomness` contract in practically all the EVM-compatible chains that are already integrated with the Witnet side-chain: 
+
+  {% content-ref url="contract-addresses.md" %}
+  [contract-addresses.md](contract-addresses.md)
+  {% endcontent-ref %}  
+
+While the `WitnetRandomness` contract internally uses an instance of the low-level Witnet Randomness Request explained on a [previous article](./randmoness-requests.md), this contract implements also the whole request/resolution asynchronous flow involved in the resolution of a data request in the context of the Witnet side-chain.
+
+Moreover, the same instance can be safely and concurrently used by any number of indepent applications running within the same EVM-chain (i.e. any of those currently supported by the Witnet side-chain). 
+
+  {% hint style="info" %}
+  Because of this, you may even have the chance to reuse randomness previously requested by others _**at no cost**_ (🎉), while having the certainty that the acquired randmoness was independently generated by the Witnet side-chain strictly after any given block number.
```

Comment:
```suggestion
  Because of this, you may even have the chance to reuse randomness previously requested by others _**at no cost**_ (🎉), while having the certainty that the acquired randomness was independently generated by the Witnet side-chain strictly after any given block number.
```